app/models/keyword.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class Keyword < ApplicationRecord
+  enum status: { processing: 'processing', succeeded: 'succeeded', failed: 'failed' }
+
+  has_many :keyword_search_entries, dependent: :destroy
+
+  validates :content, :status, presence: true
```

Comment:
should we add the validation for the `result_page_html` if the status is `succeeded` @Goose97 

```
validates :result_page_html, presence: true, if: succeeded?
```

Reply:
Nice catch. Fixed in dd3cf509e253beb0ad311d63e38a0422a2ee11ad